tests/generators/test_openai_compatible.py

```diff
@@ -98,3 +102,42 @@ def test_openai_multiprocessing(openai_compat_mocks, classname):
         with Pool(parallel_attempts) as attempt_pool:
             for result in attempt_pool.imap_unordered(generate_in_subprocess, prompts):
                 assert result is not None
+
+
+def test_validate_call_model_token_restrictions(openai_compat_mocks):
+    import lorem
+    import json
+    from garak.exception import GarakException
+
+    generator = build_test_instance(OpenAICompatible)
+    mock_url = getattr(generator, "uri", "https://api.openai.com/v1")
+    with respx.mock(base_url=mock_url, assert_all_called=False) as respx_mock:
+        mock_response = openai_compat_mocks["chat"]
+        respx_mock.post("chat/completions").mock(
+            return_value=httpx.Response(
+                mock_response["code"], json=mock_response["json"]
+            )
+        )
+        generator._call_model("test values")
+        resp_body = json.loads(respx_mock.routes[0].calls[0].request.content)
+        assert (
+            resp_body["max_tokens"] < generator.max_tokens
+        ), "request max_tokens must account for prompt tokens"
+
+        test_large_context = ""
+        while len(test_large_context.split()) < generator.max_tokens:
+            test_large_context += "\n".join(lorem.paragraph())
+        large_context_len = len(test_large_context.split())
+        with pytest.raises(GarakException) as exc_info:
+            generator._call_model(test_large_context)
+        assert "cannot be created" in str(
+            exc_info.value
+        ), "a prompt large then max_tokens must raise exception"
```

Comment:
I think this should be "a prompt larger than context_len should raise an exception"